### ISSUE TITLE
ClickHouse `SETTINGS` section is not supported

### ISSUE THREAD

**Issue by pheepa** [sqlfluff/sqlfluff#5931]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

ClickHouse has [query-level settings](https://clickhouse.com/docs/en/operations/settings/query-level)
Sqlfluff doesn't support settings section

### Expected Behaviour

Settings section should be parsed

### Observed Behaviour

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'settings
                       | allow_experimental_window_funct...'
```

### How to reproduce

Run `sqlfluff lint --dialect clickhouse filename.sql`
filename.sql: 
```sql
select t.a,t.b
from table as t
settings allow_experimental_window_functions = 1;
```

### Dialect

clickhouse

### Version

2.3.5

### Configuration

[sqlfluff]
dialect = clickhouse
sql_file_exts = .sql,.sql.j2,.dml,.ddl
templater = jinja
exclude_rules = LT08, CP03
max_line_length = 80

[sqlfluff:rules]
single_table_references = consistent
unquoted_identifiers_policy = all

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 4
indented_joins = False
indented_ctes = True
indented_using_on = True
indented_on_contents = True
indented_then = True
indented_then_contents = True
allow_implicit_indents = False
template_blocks_indent = True
skip_indentation_in = script_content
trailing_comments = before
ignore_comment_lines = True

[sqlfluff:layout:type:comma]
spacing_after = single
line_position = trailing

[sqlfluff:layout:type:binary_operator]
spacing_within = touch
line_position = leading

[sqlfluff:layout:type:select_clause]
line_position = alone

[sqlfluff:layout:type:where_clause]
line_position = alone

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = consistent

[tool.sqlfluff.templater.jinja]
apply_dbt_builtins = true

[sqlfluff:rules:aliasing.table]
aliasing = explicit

[sqlfluff:rules:aliasing.column]
aliasing = explicit

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
